### PR TITLE
Fix GarbageCollect for Hyper runtime

### LINKED ISSUE #98: Keep logs for terminated containers
Kubernetes will keep logs for terminated containers on Docker runtime. Hyper runtime should have same behavior.

### PR DESCRIPTION
**Pull request by feiskyer**
- Evict hyper Pod according to FinishTime of the Pod
- Remove dead container log symlinks
- also fix #98

### CODE REVIEW

**Comment by feiskyer** [comment]
@resouer Please help to review

File: pkg/kubelet/hyper/hyper.go [hunk 1]
```diff
@@ -1365,7 +1365,18 @@ func (r *runtime) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy) error
...
+	}
 
+	// Remove dead symlinks - should only happen on upgrade
+	// from a k8s version without proper log symlink cleanup
+	logSymlinks, _ := filepath.Glob(path.Join(r.containerLogsDir, "*.log"))
+	for _, logSymlink := range logSymlinks {
+		if _, err = os.Stat(logSymlink); os.IsNotExist(err) {
+			err = os.Remove(logSymlink)
+			if err != nil {
+				glog.Warningf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
```
Comment (resouer) [inline pkg/kubelet/hyper/hyper.go:1377]: Why warning not error?

Comment (feiskyer) [inline pkg/kubelet/hyper/hyper.go:1377]: Same as docker

**Comment by resouer** [comment]
lgtm

**Pull request merged**